tensorflow/compiler/xla/service/gpu/gpu_fusible.cc

```diff
@@ -541,5 +541,35 @@ bool IsConsumerTheOnlyNonRootUser(const HloInstruction& instr,
   });
 }
 
+size_t GetInstrCountOfFusible(const HloInstruction& instr) {
+  if (instr.opcode() != HloOpcode::kFusion) {
+    return 1;
+  } else {
+    return instr.fused_instruction_count();
+  }
+}
+
+absl::InlinedVector<HloInstruction*, 2> GetOutputsOfFusible(
+    const HloInstruction& instr) {
+  if (instr.opcode() != HloOpcode::kFusion) {
+    return {const_cast<HloInstruction*>(&instr)};
```

Comment:
Must we const-cast? Why not return a vector of const pointers?

Reply:
The return type has to align with the return type of `HloInstruction::operands()`, which is of `absl::InlinedVector<HloInstruction*, 2>`. So, returning absl::InlinedVector<const HloInstruction*, 2> will require new allocation of the vector. `const_cast` may be a bit less evil.

Let me know if you have any other ideas?

Reply:
`const_cast` seems to be technically UB if modifying methods are then called, right? Allocating 2 pointers on a stack (new vector) should be essentially free, right?

Reply:
Just did some search. It is undefined behavior if the object itself (not the pointer) is indeed const. Then, changing the object is undefined behavior through const_cast.

My previous statement was wrong though--we allocate a new vector (in the stack) anyway. So, let's remove the const_cast. Please help to take a look of the latest [commit](https://github.com/tensorflow/tensorflow/pull/48706/commits/0e0604868fcb74cda63333a6af1adfa3a38b03fb).